dnsutil/util.go

```diff
@@ -0,0 +1,78 @@
+// Package dnsutil contains higher-level methods useful with the dns
+// package.  While package dns implements the DNS protocols itself,
+// these functions are related but not directly required for protocol
+// processing.  They are often useful in preparing input/output of the
+// functions in package dns.
+package dnsutil
+
+import (
+	"strings"
+
+	"github.com/miekg/dns"
+)
+
+// AddDomain adds origin to s if s is not already a FQDN.
```

Comment:
AddOrigin.

"The zonefile convention" is specified in RFC 1035, Section "5.1. Format", regarding the '@' sign.